mam/api/api.c

```diff
@@ -730,11 +736,32 @@ void mam_api_serialize(mam_api_t const *const api, trit_t *const buffer) {
   mam_api_read_ctx_map_serialize(&api->read_ctxs, &trits);
   mam_pks_serialize(api->trusted_channel_ids, &trits);
   mam_pks_serialize(api->trusted_endpoint_ids, &trits);
+
+  if (!trits_is_null(encryption_key)) {
+    plain_text = trits_from_rep(trits_buffer_size, buffer);
+    mam_sponge_init(&sponge);
+    mam_sponge_absorb(&sponge, MAM_SPONGE_CTL_KEY, encryption_key);
+    mam_sponge_encr(&sponge, plain_text, cipher_text);
```

Comment:
Why don't we directly write into the buffer ?

Reply:
apparently you can pass to encr/decr inp=out without a problem, so yes, we should  write directly